src/D3MHub.t.sol

```diff
@@ -901,21 +898,21 @@ contract D3MHubTest is DSTest {
         assertEq(culled, 0);
     }
 
-    function testFail_no_uncull_not_culled() public {
+    function test_no_uncull_not_culled() public {
         _windSystem();
         d3mHub.cage(ilk);
 
         vat.cage();
-        d3mHub.uncull(ilk);
+        assertRevert(address(d3mHub), abi.encodeWithSignature("uncull(bytes32)", ilk), "D3MHub/not-prev-culled");
     }
 
-    function testFail_no_uncull_mcd_live() public {
+    function test_no_uncull_mcd_live() public {
         _windSystem();
         d3mHub.cage(ilk);
 
         d3mHub.cull(ilk);
 
-        d3mHub.uncull(ilk);
+        assertRevert(address(d3mHub), abi.encodeWithSignature("uncull(bytes32)", ilk), "D3MHub/no-uncull-normal-operation");
```

Comment:
In `testFail_hub_upgrade_kills_old_hub` you can change to `test_hub_upgrade_kills_old_hub` and do:

```
        // Ensure old hub revert
        (bool succeeded,) = address(d3mHub).call(abi.encodeWithSignature("exec(bytes32)", ilk));
        assert(!succeeded);
```
This will remove the last failTest.

Reply:
`assertRevert` actually implements this way. Or am I missing anything?

Reply:
Your are currently not using `assertRevert` for `testFail_hub_upgrade_kills_old_hub`.
I figured it was because there is no revert message there.
If that's the case then you can just do that instead.
Did not see an `assertRevert` version that does not check the error message.

Reply:
I tried using assertRevert but when running the test I got a failure due OutOfGas and not the missmatching error message. So like something is breaking getting outside the `call` environment. This is happening at least with Foundry.

Reply:
It's probably becuase it's calling a view functions on address(0) (we set the pool to that in that test).

Reply:
Ah yes that might explain it, but want to do some deeper check later. For now I just left that one remaining test as it was.